### ISSUE TITLE
Try to bump dmc

### ISSUE THREAD

**Issue by antonymilne** [mckinsey/vizro#905]
Context: https://github.com/mckinsey/vizro/issues/901#issuecomment-2505980839. dmc is now at 0.15. Let's _consider_ unpinning:
1. Higher priority: see if we can get same datepicker behaviour as now but hopefully with fewer bugs (maybe helps with dynamic filter @petar-qb?). Initially we just need to check feasibility of this and whether it's something we can do quickly and easily or whether it's a huge piece of work.
2. Lower priority/separate ticket: consider #318 and how we might like to integrate timepicker into vizro.

After https://github.com/mckinsey/vizro/pull/895 I think the only dmc components we have left are `dmc.Switch` and `dmc.DateRangePicker/dmc.DatePicker`.

@huong-li-nguyen are there plans to replace `dmc.Switch` with `dbc`?

**Comment by huong-li-nguyen** [comment]
Yes, the `dmc.Switch` will be replaced with the `dbc.Checklist` next 👍 To convert the DatePicker could be a bit more complex as the API has changed.

**Comment by petar-qb** [comment]
This issue depends a bit on fixing the dash persistence [bug](https://github.com/McK-Internal/vizro-internal/issues/1389). We can start working on this ticket, but the date_picker persistence will be harder to implement.
P.S. The date_picker persistence already doesn't work.

**Comment by AnnMarieW** [comment]
Hi Vizro Team,

I’m glad to hear that you’re considering upgrading to the latest version of `dash-mantine-components`. As the primary maintainer of DMC, I’m happy to assist if you decide to move forward with this project.

Upgrading from version `0.12` to the latest release does involve significant changes. In large apps, this can be quite time-consuming; however, in Vizro, it’s relatively straightforward since only one component is in use. To help, I’ve prepared a draft PR #924 for you to review.

The current DMC version offers many benefits: enhanced stability, more features, and critical bug fixes, all thanks to its foundation on Mantine V7. You can find more details in our [Roadmap](https://github.com/snehilvj/dash-mantine-components/discussions/377).

We’ve been making rapid progress on DMC, releasing updates approximately every two weeks. You can see recent release announcements in our [GitHub Discussions](https://github.com/snehilvj/dash-mantine-components/discussions?discussions_q=is%3Aopen+Release+0.).

---

### Benefits of Upgrading DMC in Vizro

As Antony mentioned in [this comment](https://github.com/mckinsey/vizro/issues/901#issuecomment-2505980839), version `0.12` remains relatively popular. However, I anticipate its usage will decline as new projects adopt the latest version.

I understand the desire to minimize dependencies, but upgrading DMC offers significant advantages for Vizro, including:

- Higher-Quality Date Pickers: The DMC date pickers in versions >=0.14 eliminate the need for workarounds currently included in Vizro. 
- Access to ALL DMC components:  This upgrade allows Vizro users to include any DMC component as custom components. Many of these components are not available in Dash Bootstrap Components or other Plotly libraries.
  - This upgrade would address Issue #901, enabling  the addition of `dmc.DateTimePicker`.
  - The [dmc.Stepper](https://www.dash-mantine-components.com/components/stepper) could be a solution for the "Multi-step Wizard" as requested in #888 .
- The `MantineProvider` makes it straightforward to style all Mantine components with a Vizro theme, which could help address  Brand Identity Management as outlined in Issue #887

---


### Draft Pull Request Summary
I’ve drafted a PR to upgrade Vizro to use DMC version `0.15.1`. Here’s a summary of the changes:

1. Mantine V7 requires React 18, which will be the default in Dash 3.0. For now, React 18 needs to be explicitly set in the app or via an environment variable.
2. Some components (e.g., date pickers) require extra CSS files. These are included as an external stylesheet in the PR, but it might make sense to move them to Vizro’s CSS folder.
3. The app layout needs to be wrapped in a `dmc.MantineProvider`.
4. Replaced the outdated date picker components with the new `DatePickerInput`. Eliminated all the workarounds for bugs in 0.12
5. Updated the JavaScript function to update both the Bootstrap and Mantine themes when switching.

---


### Next Steps
I haven’t updated any tests yet, but I’ve verified functionality by running the example app in the `dev` folder. Here’s the command I used:

```bash
hatch run examples:example dev
```

The app works with all components and the theme switcher, and it’s ready for further testing.

**Comment by AnnMarieW** [comment]
> This issue depends a bit on fixing the dash persistence [bug](https://github.com/McK-Internal/vizro-internal/issues/1389). We can start working on this ticket, but the date_picker persistence will be harder to implement. P.S. The date_picker persistence already doesn't work.

Can you say more about this bug? Is this a Vizro issue or can it be fixed in either Dash or DMC?   I can't access the internal issue.

**Comment by AnnMarieW** [comment]
Hi @petar-qb 
Good to know that you are perusing the Dash issue.  It would be nice to have the option to persist values that are updated in callbacks.

I see that the DatePicker is not persistent in the Vizro app I ran in my PR, however, the `dmc.DatePickerInput` persistence does work in a regular Dash app - you can see it on this sample app on [PyCafe](https://py.cafe/snippet/dash/v1?#c=H4sIAKH7VmcEA5VVwW7jNhD9FYG9JIAlWFbsxAZ0aJtDC2zRoLtAD1Eg0NIo4oYiVZJK4izy732kZLtGs5vdJEA8nHnz3sxw6C-s0jWxDRNdr42Lam7bsuPKCUVlpXGoSDkbcRvVXVWoxuguBEUT4BqfZ1EZcIZUTYbMLPpd9YObRX8OLvz_6LijQp1GJaUlB4tXrnwkY4VWZwVLr5JFMi_YeaEKxfs-ygPFGT07MorL0rqdJNsSOZtDUjLayc8fPgRIDaayF9UDmdJwdU8h6BqnN-EwKDsrVIQfUecF84B4BMQBULDZ6JZ8SxIRH0lS5SKOuh1FpzGPXA6U3xZsMV9cxPMUf_BFRztFLXdTcO-rtI5URfknM9B07HY9gWZKXKj_1WGFupc_VMiIOKh8s5KDd6zhKDmUMLq-rbimhg_SFXvNGFci-U4PDlPzav8YL9KN0Y8CMz-7HeHe9Uk4SZj3NXUaE0Zjx7Js9CRcG90ciSEm0gbwPDuf-L38fXNC2944n1qg7oI00URlqXhHZRnlOeZT4pYLVZYF24xYL94M6qym7XAfij1nM2bon0EY6vwSYEv8BfazsW08LUl8XJI8nyfpMkkLBaBv0QSA1e9cqxXsfqdrtCJ-nCeLVZLCNTaMbb6waQnYZgFerd1f-skfT5kMrBmrWiFrbBDb3B48jm-xSXA-idq1bJMu5zPWCfX3aGaj9RuJ-xY83hQ1YI2Q9AuyWjK_auXQDTJfYfCh8XaMRUjPfV7GXu9eZ4eYg4q3EsB5xPlO9zt2ij26_9vyxD27UxrzfhveL30q5b2y9xWfCDjQtFPW9HL5PZx4vnDM5Xuk-zjP6n9fZ6H9uH23d7gtunrwps-CBiH-p6jFHdtEc5pf0rLh83W1zprLVbaq1tvtet0sVmt-kTaF2kqhsBl5niZrPLFYLeNEI_LcLz5e3QWOWm5QXay06bgULz46S7LgktirPL9K0uRyXAEAE_9cTwtRaXO6DXjHPU1wtq6TX3V-a5VCAHqO9y9fjphGcgslGawMm10r7g2IGr-SpNjGHTmOxwCOqwQouJyt_UNh9BDI_XeM-izUZ77w4NRn6rh5GHrLG1ChsmRZKEXWxdzuVCW079vKw3qpndx5OWgbJO53unWu98KBLZS_xQAHsiz1MEPO7IS693my5KJQaPTQO60lglZr6ATMimfvTwMRJswr4UCFeYWDXY8Esf8uRKRWAEJByD4YicKzsTYkeiLz8EJ4yUKfVoV6ET3UYZQh3FpZKPb6L02osND_BwAA).  User values are maintained after a browser refresh.

**Comment by petar-qb** [comment]
> I see that the DatePicker is not persistent in the Vizro app I ran in my PR

Correct. That's a well known Vizro issue we also plan to address soon.
